client/ayon_nuke/api/plugin.py

```diff
@@ -158,26 +165,54 @@ def create_instance_node(
 
         return created_node
 
-    def set_selected_nodes(self, pre_create_data):
-        if pre_create_data.get("use_selection"):
-            self.selected_nodes = nuke.selectedNodes()
-            if self.selected_nodes == []:
-                raise NukeCreatorError("Creator error: No active selection")
+    def _get_current_selected_nodes(
+            self,
+            pre_create_data,
+            class_name: str = None,
+        ):
+        """ Get current node selection.
+
+        Arguments:
+            pre_create_data (dict): The creator initial data.
+            class_name (Optional[str]): Filter on a class name.
+
+        Returns:
+            list[nuke.Node]: node selection.
+        """
+        class_name = class_name or self.node_class_name
+        use_selection = pre_create_data.get("use_selection")
+
+        if use_selection:
+            selected_nodes = nuke.selectedNodes()
         else:
-            self.selected_nodes = []
+            selected_nodes = nuke.allNodes()
+
+        if class_name:
+            selected_nodes = [
+                node
+                for node in selected_nodes
+                if node.Class() == class_name
+            ]
+
+        if class_name and use_selection and not selected_nodes:
+            raise NukeCreatorError(f"Creator error: Select a {class_name} node.")
```

Comment:
Why the message says `Creator error:`?
```suggestion
            raise NukeCreatorError(f"Select a {class_name} node.")
```

Reply:
Not sure, it was there initially so I've kept it. But I think you're right we can remove it.